### PR TITLE
net_tools: Introduce networkd

### LINKED ISSUE #28709: Systemd-networkd support
I searched  for systemd-networkd~and just find bugs, nothing related to network management.

##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME
Suggestion: networkd

##### OS / ENVIRONMENT
I use Ubuntu. Unfortunately I can't provide much info because Ubuntu doesn't use systemd-networkd for network configuration.

##### SUMMARY
I think should have a module to manage network configuration when the system use systemd-networkd, I don know if today some distro uses it, so this feature idea is kind a long shot.

##### EXPECTED RESULTS
Use of a module to edit configuration files used by systemd-networkd.

##### ACTUAL RESULTS
The only solution seems edit the files manually. The problem with this approach is could happen syntax errors and network could become unavailable.

### PR DESCRIPTION
**Pull request by ssahani**
##### SUMMARY
This work is minimum set of configuration in aim of automating
systemd-networkd conf files.
Please see the doc section to how to configure yaml files.
More information  see https://www.freedesktop.org/software/systemd/man/systemd-networkd.service.html

##### ISSUE TYPE

- New Module Pull Request

Closes: #28709

##### COMPONENT NAME
networkd
##### ANSIBLE VERSION

```
"2.8 "
```

##### ADDITIONAL INFORMATION

```paste below
Initial WIP Version
```
Signed-off-by: Susant Sahani <susant@redhat.com>

### CODE REVIEW

File: lib/ansible/modules/net_tools/networkd.py [hunk 1]
```diff
@@ -0,0 +1,650 @@
...
+
+DOCUMENTATION = '''
+---
+module: networkd
+short_description: Configure Link, NetDev and Network for systemd-networkd
+description:
+    - Configure Link, NetDev and Network for systemd-networkd
+
+options:
+  ConfigType:
```
Comment (ryansb) [inline lib/ansible/modules/net_tools/networkd.py]: The capitalization casing doesn't match what is typical in Ansible. We heavily prefer `snake_case` to `CamelCase` for module parameters, so mixing up snake/camel options isn't a source of frustration for users.

**Comment by andreaso** [comment]
Wouldn't this new _networkd_ module rather belong in the `system/` namespace, which already contains the following system networky modules?

* firewalld
* hostname
* iptables
* ufw

**Comment by ssahani** [comment]
It does the networking conf same as  nmcli . So I thought It should be better belongs there.

**Comment by andreaso** [comment]
Good point! Guess the current namespace separation isn't without some inconsistencies.

**Comment by ssahani** [comment]
Ping ! Any feedback Please.

**Comment by maxamillion** [comment]
Hello and thank you for the pull request!

I'm curious, what's the goal of making this into a module vs creating a role and hosting it on [Ansible Galaxy](https://galaxy.ansible.com/)?

**Comment by ssahani** [comment]
any reason why it should be there not in here. We have nmcli iproute stuff also here. I am trying to figure out why you think so ?

**Comment by KellerFuchs** [comment]
@ssahani Hey, sorry your PR got kinda forgotten about.
If you are still up for getting that new module merged, I can make time for a review.  :)

**Comment by KellerFuchs** [comment]
@maxamillion I think it makes sense to have it done as a module:
- There is fairly complex logic that would be complicated (and slow) to run as a playbook or role.
- You might want to call it several times in the execution of a playbook, with different arguments.
- You might want to notify handlers based on whether the module changed anything.

**Comment by ssahani** [comment]
@KellerFuchs Sorry I don't have any time to work on this now. If you can rework and and merge that will be nice .

**Comment by KellerFuchs** [comment]
@ssahani Unfortunately, I'm also a bit short on snoons and time currently.
I will see what I can do, but if you get around to working on this before I do, feel free to reopen the PR and ping me (@KellerFuchs in a Github comment should drop me a notification)

**Pull request closed without merging**